### DISCUSSION TITLE
Segmentation sharply stops

### DISCUSSION THREAD

**Discussion by hogger23** [ultralytics/ultralytics#15029] [Q&A]
Hello everyone,
I am trying to learn how to use some ML tools on my own and I have decided that a little image segmentation project would be good to start because of the visual feedback of my results. I have decided to use Yolov8 segmentation models. 

You can find underneath three examples of results I got back. The two on the right are satisfactory. The one on the left has two instances (indicated by the red arrows) of the segmentation stopping as a flat horizontal line. It happens a lot and i dont understand why. 
The pictures I use are pictures of rock drill core. the nature of those items is that they are very one dimensional (very long and narrow, unbalanced aspect ratio).

![Segmentation stops as a flat boundary](https://github.com/user-attachments/assets/64ce0f16-99a6-4bd0-9a29-422bf0f3f911)

My main concern is to solve those flat stops in segmentation.

Would anyone have an idea on how to solve it ?

Thank you

**Comment by pderrenger** [comment]
@hogger23 thank you for sharing your experience and the detailed description of your issue. The flat horizontal lines in your segmentation results could be due to the aspect ratio of your images. YOLO models, including YOLOv8, typically resize images to a square shape during preprocessing, which can distort objects with extreme aspect ratios.

To address this, you might want to try enabling rectangular training and inference. This approach maintains the aspect ratio of your images, potentially improving segmentation accuracy for long and narrow objects like rock drill cores. Additionally, ensure you are using the latest version of the Ultralytics package, as updates often include bug fixes and performance improvements.

If the issue persists, consider augmenting your dataset with more varied examples or adjusting the model's hyperparameters to better suit your specific use case. 

Feel free to reach out if you have further questions or need additional assistance.
